api/src/main/java/org/apache/iceberg/TableScan.java

```diff
@@ -58,6 +58,15 @@
    */
   TableScan asOfTime(long timestampMillis);
 
+  /**
+   * Create a new {@link TableScan} from this scan's configuration that will override the {@link Table}'s read splitting
+   * behavior based on the incoming {@link SplitOptions}.
+   *
+   * @param splitOptions a {@link SplitOptions} containing the splitting options
+   * @return a new scan based on this with the splitting options set
+   */
+  TableScan splitOptions(SplitOptions splitOptions);
```

Comment:
What about using more generic options that are string keys and values? Something like this:

```java
    public TableScan option(String property, String value);
```

That would make this more generic so we can use options to control behavior other than splits, without adding a lot of methods to the scan API. What do you think?

Reply:
This would also make it easier to translate from Spark to Iceberg. You'd just need to add all of the options to the scan.

Reply:
Makes sense. Will do.